sphinx/tutorial/cylc/furthertopics/clock-triggered-tasks.rst

```diff
@@ -0,0 +1,179 @@
+simply running once their prerequisites are met. Clock triggered tasks,
+however, wait for a particular wall clock time to be reached in addition
+to their other prerequisites.
+
+
+Clock Triggering
+----------------
+
+When clock triggering tasks we can use different
+:ref:`offsets <tutorial-iso8601-durations>` to the cycle time as follows:
+
+.. code-block:: cylc
+
+   clock-trigger = taskname(CYCLE_OFFSET)
+
+.. note::
+
+   Regardless of the offset used, the task still belongs to the cycle from
+   which the offset has been applied.
+
+
+Example
+-------
+
+Our example suite will simulate a clock chiming on the hour.
+
+Within your ``~/cylc-run`` directroy create a new directory called
+``clock-trigger``::
+
+   mkdir ~/cylc-run/clock-trigger
+   cd ~/cylc-run/clock-trigger
+
+Paste the following code into a ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [cylc]
+       UTC mode = True # Ignore DST
+
+   [scheduling]
+       initial cycle point = TODO
+       final cycle point = +P1D # Run for one day
+       [[dependencies]]
+           [[[PT1H]]]
+               graph = bell
+
+   [runtime]
+       [[root]]
+           [[[events]]]
+               mail events = failed
+       [[bell]]
+           env-script = eval $(rose task-env)
+           script = printf 'bong%.0s\n' $(seq 1 $(cylc cyclepoint --print-hour))
+
+Change the initial cycle point to 00:00 this morning (e.g. if it was
+the first of January 2000 we would write ``2000-01-01T00Z``).
+
+We now have a simple suite with a single task that prints "bong" a number
+of times equal to the (cycle point) hour.
+
+Run your suite using::
+
+   cylc run clock-trigger
+
+Stop the suite after a few cycles using the :guilabel:`stop` button in the
+``cylc gui``. Notice how the tasks run as soon as possible rather than
+waiting for the actual time to be equal to the cycle point.
+
+
+Clock Triggering Tasks
+----------------------
+
+We want our clock to only ring in real time rather than the simulated
+cycle time.
+
+To do this, add the following lines to the ``[scheduling]`` section of
+your ``suite.rc``:
+
+.. code-block:: cylc
+
+   [[special tasks]]
+       clock-trigger = bell(PT0M)
+
+This tells the suite to clock trigger the ``bell`` task with a cycle
+offset of ``0`` hours.
+
+Save your changes and run your suite::
+
+   cylc run clock-trigger
+
+
+Results
+-------
+
+Your suite should now be running the bell task in realtime. Any cycle times
+that have already passed (such as the one defined by ``initial cycle time``)
+will be run as soon as possible, while those in the future will wait for that
+time to pass.
+
+At this point you may want to leave your suite running until the next hour
+has passed in order to confirm the clock triggering is working correctly.
+Once you are satisfied, stop your suite.
+
+By making the ``bell`` task a clock triggered task we have made it run in
+realtime. Thus, when the time caught up with the cycle time, the bell task
```

Comment:
'actual OR true OR wall-clock' preceding 'time' would be clearer than just 'time' in the former lone use given the two types of timekeeping here, since the use of 'real-time' in the previous sentence doesn't make the context of this instance fully and immediately comprehensible.